### PR TITLE
[Doc] Add example for partitioning while loading in INSERT OVERWRITE

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/en/loading/InsertInto.md [hunk 1]
```diff
@@ -318,6 +318,28 @@ MySQL > select * from insert_wiki_edit;
 Empty set (0.00 sec)
 ```
 
+:::note
+For tables that uses the list partitioning strategy, that is, `PARTITION BY value`, INSERT OVERWRITE supports creates new partitions in the destination table by specifying the value of the partition key. Existing partitions are overwritten as usual.
```
Comment (evelynzhaojie) [inline docs/en/loading/InsertInto.md:322]: that uses -- that use

File: docs/zh/loading/InsertInto.md [hunk 2]
```diff
@@ -300,6 +300,28 @@ MySQL > select * from insert_wiki_edit;
...
+
+以下示例创建了 List 分区表 `activity`，向其中导入新数据时自动创建了先前不存在的分区：
+
+```SQL
+CREATE TABLE activity (
+id INT          NOT NULL,
+dt VARCHAR(10)  NOT NULL
+) ENGINE=OLAP 
+DUPLICATE KEY(`id`)
+PARTITION BY (`id`, `dt`)
```
Comment (evelynzhaojie) [inline docs/zh/loading/InsertInto.md:314]: LIST 分区的语法应该是 PARTITION BY LIST ()

Comment (EsoragotoSpirit) [reply docs/zh/loading/InsertInto.md:314]: fixed

**Review by hellolilyliuyi — DISMISSED** [review]

File: docs/en/loading/InsertInto.md [hunk 3]
```diff
@@ -318,6 +318,28 @@ MySQL > select * from insert_wiki_edit;
 Empty set (0.00 sec)
 ```
 
+:::note
+For tables that use the `PARTITION BY column` strategy, INSERT OVERWRITE supports creates new partitions in the destination table by specifying the value of the partition key. Existing partitions are overwritten as usual.
+
+The following example creates the table `activity` with the list partitioning strategy, and creates a new partition in the table while inserting data into it:
```
Comment (evelynzhaojie) [inline docs/en/loading/InsertInto.md:324]: list partitioning

File: docs/zh/loading/InsertInto.md [hunk 4]
```diff
@@ -300,6 +300,28 @@ MySQL > select * from insert_wiki_edit;
 Empty set (0.00 sec)
 ```
 
+:::note
+对于通过 PARTITION BY value 方式分区的表，INSERT OVERWRITE 支持通过指定分区键的值在目标表上创建不存在的分区。对于已有的分区，将正常进行覆盖写。
```
Comment (evelynzhaojie) [inline docs/zh/loading/InsertInto.md:304]: PARTITION BY columns 英文这样写的

File: docs/zh/loading/InsertInto.md [hunk 5]
```diff
@@ -300,6 +300,28 @@ MySQL > select * from insert_wiki_edit;
...
+
+以下示例创建了 List 分区表 `activity`，向其中导入新数据时自动创建了先前不存在的分区：
```
Comment (evelynzhaojie) [inline docs/zh/loading/InsertInto.md:306]: 不是 List 分区表，建议：以下示例创建了按照 `id` 和 `dt` 列进行分区的表 `activity`

File: docs/zh/loading/InsertInto.md [hunk 6]
```diff
@@ -300,6 +300,28 @@ MySQL > select * from insert_wiki_edit;
 Empty set (0.00 sec)
 ```
 
+:::note
+对于通过 PARTITION BY column 方式分区的表，INSERT OVERWRITE 支持通过指定分区键的值在目标表上创建不存在的分区。对于已有的分区，将正常进行覆盖写。
```
Comment (hellolilyliuyi) [inline docs/zh/loading/InsertInto.md:304]: PARTITION BY column 是什么？

Comment (EsoragotoSpirit) [reply docs/zh/loading/InsertInto.md:304]: Partition by <cloumn_name>

Comment (hellolilyliuyi) [reply docs/zh/loading/InsertInto.md:304]: 直接跟下面一样写list分区的表是不是就可以？

File: docs/zh/loading/InsertInto.md (hunk 2 above)
Comment (hellolilyliuyi) [reply docs/zh/loading/InsertInto.md:314]: 我这边看到还是

**Review by evelynzhaojie — DISMISSED** [review]

File: docs/zh/loading/InsertInto.md (hunk 2 above)
Comment (EsoragotoSpirit) [reply docs/zh/loading/InsertInto.md:314]: 这里不是 List 分区。

**Review by evelynzhaojie — APPROVED** [review]

**Review by hellolilyliuyi — APPROVED** [review]

**Pull request merged**